### ISSUE TITLE
[Webpack 4 + Next.js]: `wallets` is undefined in `WalletsModal.mjs`

### ISSUE THREAD

**Issue by marianlam** [solana-labs/wallet-adapter#322]
### **Describe the bug**

**TLDR:** Incompatibility with Webpack 4.

`wallets` is undefined in `WalletsModal.mjs` even though it should have been initialized earlier when `WalletProvider` code was executed. Note that in this situation, I followed the docs closely to ensure components are wrapped within the necessary context providers.

### **To Reproduce**
1. Create a Next.js app (using typescript in my case, but that might not be necessary to repro)
`yarn create next-app --typescript`
2. Configure Webpack with the following in `next.config.js`:
```
module.exports = {
  // Webpack 5 is enabled by default
  // You can still use webpack 4 while upgrading to the latest version of Next.js by adding the "webpack5: false" flag
  webpack5: false,
  webpack: (config) => {
    config.module.rules.push({
      test: /\.mjs$/,
      include: /node_modules/,
      type: 'javascript/auto',
    });
    return config;
  },
};
```
3. Follow instructions in starter documentation to setup wallet-adapter in Next.js app, compile with Webpack 4 and deploy
4. Click on the "Select Wallet" (`<WalletMultiButton />`) button
5. See error

```
Unhandled Runtime Error
TypeError: wallets is not iterable

Call Stack
eval
node_modules/@solana/wallet-adapter-react-ui/lib/esm/WalletModal.mjs (20:0)
mountMemo
node_modules/react-dom/cjs/react-dom.development.js (15846:0)
Object.useMemo
node_modules/react-dom/cjs/react-dom.development.js (16219:0)
useMemo
node_modules/react/cjs/react.development.js (1532:0)
WalletModal
node_modules/@solana/wallet-adapter-react-ui/lib/esm/WalletModal.mjs (16:52)
renderWithHooks
node_modules/react-dom/cjs/react-dom.development.js (14985:0)
mountIndeterminateComponent
node_modules/react-dom/cjs/react-dom.development.js (17811:0)
beginWork
node_modules/react-dom/cjs/react-dom.development.js (19049:0)
HTMLUnknownElement.callCallback
node_modules/react-dom/cjs/react-dom.development.js (3945:0)
Object.invokeGuardedCallbackDev
node_modules/react-dom/cjs/react-dom.development.js (3994:0)
invokeGuardedCallback
node_modules/react-dom/cjs/react-dom.development.js (4056:0)
beginWork$1
node_modules/react-dom/cjs/react-dom.development.js (23964:0)
performUnitOfWork
node_modules/react-dom/cjs/react-dom.development.js (22776:0)
workLoopSync
node_modules/react-dom/cjs/react-dom.development.js (22707:0)
renderRootSync
node_modules/react-dom/cjs/react-dom.development.js (22670:0)
performSyncWorkOnRoot
node_modules/react-dom/cjs/react-dom.development.js (22293:0)
eval
node_modules/react-dom/cjs/react-dom.development.js (11327:0)
unstable_runWithPriority
node_modules/scheduler/cjs/scheduler.development.js (468:0)
runWithPriority$1
node_modules/react-dom/cjs/react-dom.development.js (11276:0)
flushSyncCallbackQueueImpl
node_modules/react-dom/cjs/react-dom.development.js (11322:0)
flushSyncCallbackQueue
node_modules/react-dom/cjs/react-dom.development.js (11309:0)
discreteUpdates$1
node_modules/react-dom/cjs/react-dom.development.js (22420:0)
discreteUpdates
node_modules/react-dom/cjs/react-dom.development.js (3756:0)
dispatchDiscreteEvent
node_modules/react-dom/cjs/react-dom.development.js (5889:0)
```

### **Expected behavior**
No error. `wallets` should have been initialized to be a list of `Adapters`. 

### **Screenshots**
<img width="1185" alt="image" src="https://user-images.githubusercontent.com/17263324/154793940-4e1ac8a0-a75b-4b0b-91db-a68d57580a93.png">

### **Desktop (please complete the following information):**
 - OS: macOS Big Sur (version 11.0.1)
 - Browser: Chrome
 - Version: 98.0.4758.102 (Official Build) (arm64)

### **Additional Notes**
Since this is a Webpack issue, I'm wondering if it's possible to make changes on your end to make this library compatible with Webpack 4, since I have another dependency in my project that prevents me from using Webpack 5. 🥲 If it's not a quick fix / priority, then please feel free to close out. I'll just need to find another hack to get my project working with wallet-adapter if not at all.

**Comment by jordansexton** [comment]
Webpack 4 won't be supported, sorry!

- Could you use one of the starter projects with Parcel or Next.js if using Webpack 5 is out of the question?
- Could you import the .js/CJS files instead of .mjs/ESM files?
- What's the specific dependency that you can't update Webpack for?

**Comment by jordansexton** [comment]
> 1. Create a Next.js app (using typescript in my case, but that might not be necessary to repro)
>        `yarn create next-app --typescript`

I'm confused by this, because the [nextjs-starter](https://github.com/solana-labs/wallet-adapter/tree/master/packages/starter/nextjs-starter) package was created using exactly this command and requires no additional Webpack config to support the ESM build.

**Comment by marianlam** [comment]
@jordansexton Building a wallet-adapter starter project using the defaults for `yarn create next-app --typescript` works as expected. Again, it's just I had to downgrade to using webpack 4 in my case to support a mission critical library. That's where the compilation errors start appearing. 

Anyhow, thank you for confirming! Going to close this since there's nothing actionable on your end. I'm thinking about splitting up my code into separate projects to get around this webpack version dillemma.

**Comment by asktree** [comment]
I've recently started getting the same error even though I don't use webpack 4
